infrahub_sdk/node/relationship.py

```diff
@@ -156,8 +159,29 @@ async def fetch(self) -> None:
             self.peers = rm.peers
             self.initialized = True
 
+        ids_per_kind_map = defaultdict(list)
         for peer in self.peers:
-            await peer.fetch()  # type: ignore[misc]
+            if not peer.id or not peer.typename:
+                raise Error("Unable to fetch the peer, id and/or typename are not defined")
+            if peer.typename not in ids_per_kind_map:
```

Comment:
with the defaultdict you can just do 
```
ids_per_kind_map[peer.typename].append(peer.id)
```

Reply:
Thanks :sweat_smile:  will try to include this in another PR